syntax = 'proto2';
package reporting;

service Info {
  rpc GetInfo(Coords) returns (stream Ship) {} ;
}

message Coords {
  required float ra_hours = 1;
  required float ra_minutes = 2;
  required float ra_seconds = 3;
  required float dec_degrees = 4;
  required float dec_minutes = 5;
  required float dec_seconds = 6;
}

message Ship {
  required Alignment alias = 1;
  optional string name = 2 [default = "Unknown"];
  required float length = 3;
  required Class ship_class = 4;
  required int64 crew_size = 5;
  required bool arm_status = 6;
  repeated Officers officers = 7;

  message Officers {
    required string first_name = 1;
    required string last_name = 2;
    required string rank = 3;
  }
}

enum Alignment {
  Ally = 0;
  Enemy = 1;
}

enum Class {
  Corvette = 0;
  Frigate = 1;
  Cruiser = 2;
  Destroyer = 3;
  Carrier = 4;
  Dreadnought = 5;
}
